gui/default/syncthing/core/syncthingController.js

```diff
@@ -1578,14 +1578,16 @@ angular.module('syncthing.core')
         }
 
         $scope.editDeviceModalTitle = function() {
-            if ($scope.editingDefaults) {
-                return $translate.instant("Edit Device Defaults");
-            }
             var title = '';
-            if ($scope.editingExisting) {
+            switch ($scope.currentDevice._editing) {
+            case "existing":
                 title += $translate.instant("Edit Device");
-            } else {
+                break;
+            case "add":
                 title += $translate.instant("Add Device");
+                break;
+            case "defaults":
+                return $translate.instant("Edit Device Defaults");
```

Comment:
I don't get what this changes? `title` is empty at this point.

Reply:
All other cases do not assume that the title is initially empty. Only this case would break if someone were to add something to the initial definition.

Reply:
And the suggested change was also to include a comment in the line returning.  It's easy to miss that it won't execute the rest of the function.

Reply:
I moved this condition to the beginning of the function as before the PR - seems the least missable.